packages/blocks/src/api/utils.js

```diff
@@ -43,20 +43,27 @@ export function isUnmodifiedBlock( block, role ) {
 	const blockAttributes = getBlockType( block.name )?.attributes ?? {};
 
 	// Filter attributes by role if a role is provided.
-	const attributesToCheck = role
+	const attributesByRole = role
 		? Object.entries( blockAttributes ).filter( ( [ key, definition ] ) => {
 				// A special case for the metadata attribute.
 				// It can include block bindings that serve as a source of content,
 				// without directly modifying content attributes.
 				if ( role === 'content' && key === 'metadata' ) {
-					return true;
+					return (
+						Object.keys( block.attributes[ key ]?.bindings ?? {} )
```

Comment:
Totally optional, but could this be checking `definition?.bindings ?? {}` since we already grab the value in this loop?

Reply:
Ah, of course, thanks! My drive-by comment was too quick and I hadn't double-checked the logic.

Sounds good!